### PR TITLE
Rename `type_ids` table and `version_id` column

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

We use the term `type_id` when referring to a versioned URI. Similarly, the term `version_id` is confusing. This PR renames the `type_ids` table to `ontology_ids` and the `version_id` column to `ontology_id`.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1202805690238892/1203214689883089/f) _(internal)_

## 🔍 What does this change?

- Rename `type_ids` to `ontology_ids`
- Rename `ontology_ids::version_id` to `ontology_ids::ontology_id`

## 📜 Does this require a change to the docs?

This is an internal change only

### CODE REVIEW

File: apps/hash-graph/lib/graph/src/store/postgres/ontology/ontology_id.rs [hunk 1]
```diff
@@ -5,15 +5,12 @@ use serde::{Deserialize, Serialize};
 use utoipa::ToSchema;
 use uuid::Uuid;
 
-// TODO: rename this to TypeInternalVersionId or something to distinguish it from versioned
-//  URIs and from entity ids
-//  https://app.asana.com/0/1202805690238892/1203214689883089/f
```
Comment (thehabbos007) [inline apps/hash-graph/lib/graph/src/store/postgres/ontology/ontology_id.rs:10]: I didn't even know we had this task when you talked about doing this. Great thing we already wanted to change this!

File: apps/hash-graph/postgres_migrations/V2__ontology_tables.sql [hunk 2]
```diff
@@ -1,11 +1,11 @@
 CREATE TABLE IF NOT EXISTS
-  "type_ids" (
-    "version_id" UUID PRIMARY KEY,
+  ontology_ids (
+    ontology_id UUID PRIMARY KEY,
```
Comment (thehabbos007) [inline apps/hash-graph/postgres_migrations/V2__ontology_tables.sql:3]: small nit, does our formatter not ensure quoting/non-quoting for table/column names? I do like to quote column names as it doesn't get strange highlighting overlaps with builtin keywords.

**Review by thehabbos007 — COMMENTED** [review]
Looks good to me, Tim! I just have one comment in the migration files

File: apps/hash-graph/lib/graph/src/store/postgres/ontology/ontology_id.rs (hunk 1 above)
Comment (TimDiekmann) [reply apps/hash-graph/lib/graph/src/store/postgres/ontology/ontology_id.rs:10]: Me neither, I found the TODO and was happy to remove it 😆

File: apps/hash-graph/postgres_migrations/V2__ontology_tables.sql (hunk 2 above)
Comment (TimDiekmann) [reply apps/hash-graph/postgres_migrations/V2__ontology_tables.sql:3]: The formatter does not really work as expected. I fixed this in https://github.com/hashintel/hash/pull/1975, but I'm happy to do this in this PR as well.

Comment (thehabbos007) [reply apps/hash-graph/postgres_migrations/V2__ontology_tables.sql:3]: No worries, I'm fine with the change being in that PR instead :+1:

**Review by thehabbos007 — APPROVED** [review]

**Pull request merged**